packages/shared/src/hooks/post/usePostShareLoop.ts

```diff
@@ -50,24 +62,39 @@ export const usePostShareLoop = (post: Post): UsePostShareLoop => {
     },
   });
 
+  const linkData = queryClient.getQueryData<{
+    getShortUrl: string;
+  }>(linkKey);
+
+  useEffect(() => {
+    if (linkData) {
+      setHasCopied(true);
+    }
+  }, [linkData]);
```

Comment:
I don't think you need a useEffect here right?
Could just evaluate `!!linkData`?